feat(MMENG-4169): add publish-to-mrrc task

ligangty opened the pull request:
This commit adds a task to publish artifacts from maven zip to MRRC(maven.repository.redhat.com)

ligangty committed:
feat(MMENG-4169): add publish-to-mrrc task

This commit adds a task to publish artifacts from maven zip
to MRRC(maven.repository.redhat.com)

Signed-off-by: Gang Li <ligangty@users.noreply.github.com>

ligangty commented:
Hi @mmalina @johnbieren, this is the following task of https://github.com/konflux-ci/release-service-catalog/pull/493, I'm still preparing the tests for this, but I want to submit it as draft here to start the review to see if any early issues here.

And I've one question for the tests of this task.  You will see that this task used "quay.io/redhat-services-prod/spmm-charon-containe-tenant/charon" image. It's a dedicated image to do the MRRC publish work and built and released through Konflux p02 prod cluster. The question here is how can I "mock" this image without a read token in test?

johnbieren commented on tasks/publish-to-mrrc/README.md:
```diff
@@ -0,0 +1,12 @@
+# collect-mrrc-params
```
should be `publish-to-mrrc`

ligangty replied:
Oops, copy-paste errors. Sorry.

ligangty replied:
Fixed

johnbieren commented on tasks/publish-to-mrrc/README.md:
```diff
@@ -0,0 +1,12 @@
+
+| Name | Description | Optional | Default value |
+|------|-------------|----------|---------------|
+| mrrcParamFilePath | path of the mrrc.env file which contains the MRRC parameters as environment viariables | No | |
```
can you capitalize the first character of each description string?

ligangty replied:
Will fix.

ligangty replied:
Fixed.

johnbieren commented on tasks/publish-to-mrrc/README.md:
```diff
@@ -0,0 +1,12 @@
+## Parameters
+
+| Name | Description | Optional | Default value |
+|------|-------------|----------|---------------|
```
Sean just went through and standardized all of our READMEs here https://github.com/konflux-ci/release-service-catalog/pull/509/files
As such, can you add the spaces here so the following lines line up vertically? Here is an example https://raw.githubusercontent.com/seanconroy2021/release-service-catalog/b30be07186ee04d50ee5eefc45eb32c66150723d/tasks/apply-mapping/README.md

ligangty replied:
Sure. Will fix.

ligangty replied:
Fixed.

johnbieren replied:
It looks a lot closer now, but not exactly there. It looks like the name fields are off by 1 character and the descriptions are kind of all over the place

ligangty replied:
Um... Seems I can not use vscode to do this alignment as it's totally mal-formatted. I've changed again use some other alignment way.

ligangty replied:
Fixed in new commit.

johnbieren replied:
thank you!

johnbieren commented on tasks/publish-to-mrrc/publish-to-mrrc.yaml:
```diff
@@ -0,0 +1,70 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
```
Can you add the labels and annotations like https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/add-fbc-contribution/add-fbc-contribution.yaml#L6?

ligangty replied:
Sure, will fix.

ligangty replied:
Fixed.

johnbieren commented on tasks/publish-to-mrrc/publish-to-mrrc.yaml:
```diff
@@ -0,0 +1,70 @@
+      type: string
+  steps:
+    - name: verify-repo-signature
+      image: registry.redhat.io/rhtas/cosign-rhel9:1.0.2
```
please use a sha tag. We don't like floating tags in tasks because we have no guarantee we get the same one each time and a bad push to the floating tag could break users

ligangty replied:
Sure will fix.

ligangty replied:
Fixed. Use a sha256 tag now.

ligangty commented:
> > Hi @mmalina @johnbieren, this is the following task of #493, I'm still preparing the tests for this, but I want to submit it as draft here to start the review to see if any early issues here.
> > And I've one question for the tests of this task. You will see that this task used "quay.io/redhat-services-prod/spmm-charon-containe-tenant/charon" image. It's a dedicated image to do the MRRC publish work and built and released through Konflux p02 prod cluster. The question here is how can I "mock" this image without a read token in test?
> 
> I'm not fully getting what you mean by mocking the image. It is the image the step runs with so the pod is running. What do you mean mock it? We mock binaries, but we don't modify the task step images

The main problem is this "quay.io/redhat-services-prod/spmm-charon-containe-tenant/charon" is a private registry which needs token to access. I'm not sure how to make the "tekton tests" env have permission to access this image, so that's why I'm asking if this part can be "mock" or some alternative way to handle?

mmalina commented:
> > > Hi @mmalina @johnbieren, this is the following task of #493, I'm still preparing the tests for this, but I want to submit it as draft here to start the review to see if any early issues here.
> > > And I've one question for the tests of this task. You will see that this task used "quay.io/redhat-services-prod/spmm-charon-containe-tenant/charon" image. It's a dedicated image to do the MRRC publish work and built and released through Konflux p02 prod cluster. The question here is how can I "mock" this image without a read token in test?
> > 
> > 
> > I'm not fully getting what you mean by mocking the image. It is the image the step runs with so the pod is running. What do you mean mock it? We mock binaries, but we don't modify the task step images
> 
> The main problem is this "quay.io/redhat-services-prod/spmm-charon-containe-tenant/charon" is a private registry which needs token to access. I'm not sure how to make the "tekton tests" env have permission to access this image, so that's why I'm asking if this part can be "mock" or some alternative way to handle?

You could in theory replace the image reference before running the test, but I don't think that's a good idea - then you would be testing with a completely different image, so the value of the test would be pretty low. Is there anything private about charon? Could it instead be added to [our release-service-utils image](https://github.com/konflux-ci/release-service-utils/blob/main/Dockerfile)?

mmalina commented on tasks/publish-to-mrrc/publish-to-mrrc.yaml:
```diff
@@ -0,0 +1,75 @@
+      type: string
+  steps:
+    - name: verify-repo-signature
+      image: registry.redhat.io/rhtas/cosign-rhel9@sha256:75791f7ee9257e837d1e3db13415d858c397e803ffa4534c51ca19a2ba34b2ab
```
Could you use the release-service-utils image? It has cosign in it. Or is there a reason this other image is used?

ligangty replied:
I've tried to use the latest tag of release-service-utils in recent local testing, but it failed to do verification of my testing oci-artifact. Seems it is a cosign version issue. The cosign in release-service-utils is using this version:
```
GitVersion:    v2.1.1
GitCommit:     baf97ccb4926ed09c8f204b537dc0ee77b60d043
GitTreeState:  clean
BuildDate:     2023-06-27T06:57:11Z
GoVersion:     go1.20.5
Compiler:      gc
Platform:      linux/amd64
```

And the one in rhtas/cosign-rhel9  is using this one:
```
GitVersion:    54ada3ad
GitCommit:     
GitTreeState:  
BuildDate:     
GoVersion:     go1.21.3
Compiler:      gc
Platform:      linux/amd64
```
Maybe we should upgrade the cosign in release-service-utils to more recent version?

mmalina replied:
Yes, I would suggest to update the utils image to the newer cosign version. Hopefully then it will work for you.

ligangty replied:
Should I open a jira issue for this?

mmalina replied:
You can just submit a PR to bump the version and explain that it's related to this PR and your jira in the description.

johnbieren replied:
https://github.com/konflux-ci/release-service-utils/blob/main/Dockerfile#L4 should be the only line you have to change in the PR

ligangty replied:
Ok. PR submitted here: https://github.com/konflux-ci/release-service-utils/pull/215

johnbieren replied:
The tag for your new release-service-utils image is `4576e1e2a30439129cb69c8a4f39f7ced2d44376`

ligangty replied:
Updated the image to this one and shrink the first two steps into one with this image.
Thanks for the info!

mmalina commented on tasks/publish-to-mrrc/publish-to-mrrc.yaml:
```diff
@@ -0,0 +1,75 @@
+        - name: "pub-key-vol"
+          mountPath: "/home/konflux/.cosign/"
+    - name: download-maven-repo
+      image: quay.io/konflux-ci/oras:477ee20f7e6d146ea203c4aaa0507f4d1e925a53
```
Could you use the release-service-utils image? It has oras in it.

ligangty replied:
Ok will use.

ligangty replied:
Fixed

mmalina replied:
Cool, thanks.

mmalina commented on tasks/publish-to-mrrc/publish-to-mrrc.yaml:
```diff
@@ -0,0 +1,75 @@
+      secret:
+        secretName: "$(params.charonAWSSecret)"
+  workspaces:
+    - name: data
```
Missing new line at the end.

ligangty replied:
Sure will fix.

ligangty replied:
Fixed

ligangty commented:
> > > > Hi @mmalina @johnbieren, this is the following task of #493, I'm still preparing the tests for this, but I want to submit it as draft here to start the review to see if any early issues here.
> > > > And I've one question for the tests of this task. You will see that this task used "quay.io/redhat-services-prod/spmm-charon-containe-tenant/charon" image. It's a dedicated image to do the MRRC publish work and built and released through Konflux p02 prod cluster. The question here is how can I "mock" this image without a read token in test?
> > > 
> > > 
> > > I'm not fully getting what you mean by mocking the image. It is the image the step runs with so the pod is running. What do you mean mock it? We mock binaries, but we don't modify the task step images
> > 
> > 
> > The main problem is this "quay.io/redhat-services-prod/spmm-charon-containe-tenant/charon" is a private registry which needs token to access. I'm not sure how to make the "tekton tests" env have permission to access this image, so that's why I'm asking if this part can be "mock" or some alternative way to handle?
> 
> You could in theory replace the image reference before running the test, but I don't think that's a good idea - then you would be testing with a completely different image, so the value of the test would be pretty low. Is there anything private about charon? Could it instead be added to [our release-service-utils image](https://github.com/konflux-ci/release-service-utils/blob/main/Dockerfile)?

For charon itself it's ok to release to a public repo which is in "konflux-ci" org, as it is an opensource project here: https://github.com/Commonjava/charon. But because the charon container should contain a internal tool "rpm-sign" and our internal configuration for kerberos in /etc/krb5.conf for signing purpose, I don't know if it is good idea to make it public.
We host a repo "nos/charon-containers" in our internal gitlab.cee, which is used to publish charon container through Konflux to the quay.io/redhat-services-prod, if you are interested you can see what it is containing there.

mmalina commented on tasks/publish-to-mrrc/publish-to-mrrc.yaml:
```diff
@@ -0,0 +1,75 @@
+      description: the secret name for charon aws credential file
+      type: string
+  steps:
+    - name: verify-repo-signature
```
If all the steps end up using the same image in the end (the release-service-utils image), maybe it can all be done in one step? It's ok to have more steps if it helps with readability or debugging of issues. But here it seems that it's pretty small overall (once the duplicate things are removed) and maybe having multiple steps does not have much benefit and it's just extra overhead to spin up two extra containers. WDYT?

ligangty replied:
If the images can be consolidated together, I agree to use single step to do this which makes it more readable. But I'm really not sure if the charon image is easy to be integrated into release-service-utils. The reason is discussed in following thread.

ligangty replied:
@mmalina I think we agreed to keep the charon image separated, so is this thread ok to resolve?

mmalina replied:
Yep, thanks.

mmalina commented on tasks/publish-to-mrrc/README.md:
(Starting a thread on the file, so that we can continue discussing in a thread instead of using separate comments.)

> For charon itself it's ok to release to a public repo which is in "konflux-ci" org, as it is an opensource project here: https://github.com/Commonjava/charon. But because the charon container should contain a internal tool "rpm-sign" and our internal configuration for kerberos in /etc/krb5.conf for signing purpose, I don't know if it is good idea to make it public.
> We host a repo "nos/charon-containers" in our internal gitlab.cee, which is used to publish charon container through Konflux to the quay.io/redhat-services-prod, if you are interested you can see what it is containing there.

Hmm, I'm not sure what the best solution for this is. I was gonna suggest to copy the charon binary in our Dockerfile, like we do for oras: https://github.com/konflux-ci/release-service-utils/blob/main/Dockerfile#L19
But the build pipeline of our utils image wouldn't likely have access to that image either... So maybe it could be installed in the Dockerfile directly from the github repo?

ligangty replied:
It's hard to directly copy charon to release-service-utils, because the charon tool is implemented using python, so it is not a single binary but with lots of other python deps. And another point is the internal tool "rpm-sign" and its internal configuration with /etc/krb5.conf.
I agree this container is accessible in the final managed workspace (rhtap-releng), the current problem is that how can I implement the tekton tests here, because I don't know if the test env has accessible to this image.

mmalina replied:
This is the first time we're dealing with such a thing. Until now, the referenced images were always public. I would really prefer if we can keep it that way. The task testing is one big reason for that. I guess if there's no other option, we'll have to do it. @johnbieren thoughts?

> the current problem is that how can I implement the tekton tests here, because I don't know if the test env has accessible to this image.

Yeah, I don't think it would work. We don't have any pull secrets set up for the CI in this repo. And also, we want to enable people to run the tests locally, so that would complicate things there as well.

johnbieren replied:
Yeah, unfortunately this is the first time we've tried using a private image in a release pipeline task. I don't think we've really considered it before so I don't know what the best path forward is. @davidmogar do you have any input on this?

johnbieren replied:
Just as an update, David is on PTO til Monday, but I think we should wait for his input

davidmogar replied:
@ligangty I'm assuming that both, the rpm-sign tool and the kerberos file are needed for this task. If that's the case, can the internal tool (rpm-sign) be added to a public image? If so, the only concern would be the kerberos file and it could be made accessible through a secret in the task pod.

As it was mentioned above, keeping our images public would be best. Also, keeping a single image across our tasks increases the overall performance as we need less pulls.

ligangty replied:
@davidmogar from what I see, the "rpm-sign" tool is installed through an internal rpm repo, but it's just a single binary executable file which is compiled from a .c file. The only concern is that this c file includes an url pointing to an internal key server. I'm not sure if this is sensitive to be exposed to public. If it is not so sensitive, I agree to make it public. And I agree to make that krb file as mountable through configmap or secret.

johnbieren replied:
This uses a public image now so I am resolving this

johnbieren commented on tasks/publish-to-mrrc/publish-to-mrrc.yaml:
```diff
@@ -0,0 +1,75 @@
+        - name: "pub-key-vol"
+          mountPath: "/home/konflux/.cosign/"
+    - name: download-maven-repo
+      image: quay.io/konflux-ci/release-service-utils:2a7815cc4bed65918422cf3c9fab4d2d3f298f58
```
I know this is a commit based tag, but it still isn't `@sha256`. Can you use one like that? Same for line 47

ligangty replied:
Um...I'm not sure what's different, because I see many other tasks are also using this type of tag. Whatever, I'll update to digest based tag like the cosign one.

mmalina replied:
It is different, because the digest reference really tells you which image it is. A tag can always change in theory (but not really for these sha based tags).

You're right, it's actually the convention of this repo to use the sha tag (and not the `@sha` reference). @johnbieren it should be ok, right?

johnbieren replied:
Yes, I thought we used sha tags throughout, but @ligangty is right. Resolving this, sorry for the noise

johnbieren commented on tasks/publish-to-mrrc/publish-to-mrrc.yaml:
```diff
@@ -0,0 +1,75 @@
+        name: "$(params.cosignPubKeyConfig)"
+    - name: "charon-configs-vol"
+      configMap:
+        name: "charon-configs"
```
Have we thought about how this configMap is going to end up on the konflux clusters?

ligangty replied:
I think this needs help from admins in rhtap-releng workspace to set it up. But I'm also considering if we should directly set it up in this task or in RPA as it's a yaml file in this configmap. Not sure which way is better from Konflux POV.

mmalina replied:
> But I'm also considering if we should directly set it up in this task

You mean hardcode the settings in the task? I'd say the configMap sounds probably better, yeah. And if it's supposed to be the same for everybody, having it in RPA doesn't sound like a good fit. But I wonder what Johnny thinks.

johnbieren replied:
I am curious about this. Do the values just change for stage vs prod? Does something else make them change? We do things like this https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/push-disk-images/push-disk-images.yaml#L45 in other tasks, which sounds like it could work here. Would that work here? Avoiding the configMap would be nice because we need a way to deploy it on all of our clusters + we will have to deploy it on the ephemeral clusters that run e2e tests using this task

ligangty replied:
A sample of this "charon-configs" is like this one: https://github.com/Commonjava/charon/blob/main/config/charon.yaml.sample
This can be changed by user, but if no exception in the final managed workspace, it will be a fixed content whatever the environment will be. But the content may be a little large, so I'm not sure if it's a good idea to hardcode it into task itself or a RPA. (And it may contain some aws bucket info which may be a little sensitive but not so much)
And another idea is storing this config file in the internal gitlab and give a url of it into RPA, then download it in this task. This will not need the configmap anymore. WDYT?

ligangty replied:
And another idea is storing this config file in the internal gitlab and give a url of it into RPA, then download it in this task. This will not need the configmap anymore. WDYT?

mmalina replied:
I'm fine with having it in a configMap. I mean there is still no standardized way of managing configMaps and secrets that are needed by the release pipeline which is not great. But I heard that they will use RemoteSecrets in the future. Not sure about configMaps though. See also my comment here: https://github.com/konflux-ci/release-service-catalog/pull/510#discussion_r1745293693

johnbieren replied:
My issue with the configMap is what is creating it? We don't migrate clusters daily or something, but it does happen somewhat regularly. When we migrate, what if @ligangty is on PTO? Who creates this configMap? Do they know what to put in it? I don't really. Any manual step after provisioning a new cluster like creating a configMap is not too good

ligangty replied:
The charon config yaml file will be changed in different environment(dev vs stage vs prod). I think the concerns with configmap make sense. I'll try to implement based on the RPA solution. But for the secret "awsSecret" here, I really don't know how to use some other ways except the k8s secret now in Konflux. Do you have some recommendations?

johnbieren replied:
We have different environments for lots of things, including secrets. Like here is an example https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/push-disk-images/push-disk-images.yaml#L46 and here https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/create-advisory/create-advisory.yaml#L84 you can see us checking the env by what image styles are provided. As far as secrets, https://redhat-internal.slack.com/archives/C031USXS2FJ/p1725539381988949 maybe

mmalina replied:
> As far as secrets, https://redhat-internal.slack.com/archives/C031USXS2FJ/p1725539381988949 maybe

The dist of it is that in the future this will be managed via RemoteSecrets, but for now secrets are added manually by Scott or Ralph as needed.

So I believe for now the secret is fine the way you have it.

ligangty replied:
@johnbieren @mmalina For the configmap issue, I'd like to do a separate PR later after this PR because it also needs to change the "collect-mrrc-params" together to accommodate the function. So can we close this thread in this PR?

johnbieren replied:
That's fine with me. Resolving

ligangty commented:
Hi @johnbieren @mmalina @davidmogar After discussion, we decided to publish the charon image to public for first tech preview version. I've changed the task to use the public one, and added the test pipelines. Now it's ready for review.

mmalina commented on tasks/publish-to-mrrc/README.md:
@ligangty can you look at the checkton issues please? https://github.com/konflux-ci/release-service-catalog/actions/runs/10701575099/job/29667817965?pr=510
Most of them are just about wrapping variables in `""`.

ligangty replied:
Ok will fix.

ligangty replied:
@mmalina Fixed the all quote wrapping issues there in new commit. There are two left issues  of "can't follow non-constant source", I don't know how to fix because it needs the "constant" source which is fixed path, but we don't have it there as it is derived from "${workspace.data.path}". Do you have recommendations there?

ligangty replied:
Ok, just found the recommendation to point it to /dev/null. So fixed as proposal.

mmalina replied:
Yep, makes sense, thanks.

mmalina commented on tasks/publish-to-mrrc/README.md:
```diff
@@ -0,0 +1,12 @@
+# publish-to-mrrc
+
+Tekton task publish the maven artifacts to MRRC(maven.repository.redhat.com) service. MRRC is used to host maven artifacts of Red Hat Middleware products.
```
Typo: Tekton task to publish...?

ligangty replied:
Oh sorry about the typo. Will fix.

ligangty replied:
Fixed in new commit.

mmalina replied:
There is another typo now ;)

publishs -> publishes

ligangty replied:
Oh, sorry. Fixed in new commit.

mmalina commented on tasks/publish-to-mrrc/publish-to-mrrc.yaml:
```diff
@@ -0,0 +1,59 @@
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  params:
```
The `cosignPubKeyConfig` parameter is missing? It is mentioned in the README. But now I see that the value is hardcoded under `volumes`. I would recommend adding the parameter with a default value of what you have - I think it's nicer than hardcoding it inside the task. WDYT?

And now I see that we had a discussion related to this here: https://github.com/konflux-ci/release-service-catalog/pull/510/files#r1717348622

ligangty replied:
Oh. I forgot to remove this in README, because it will not be needed anymore per recent requirement changes. This cosign pub key is needed to verify the signature of the oci-artifact, in new requirements it will be verified by existing ECP verification task like the other pipelines. I'll remove it from README

ligangty replied:
Removed in new commit.

mmalina commented on tasks/publish-to-mrrc/publish-to-mrrc.yaml:
```diff
@@ -0,0 +1,59 @@
+    - name: prepare-repo
+      image: quay.io/konflux-ci/release-service-utils:4576e1e2a30439129cb69c8a4f39f7ced2d44376
+      script: |
+        set -eux
```
Can you add a shebang? It's not strictly required, but including it makes it more explicit and also consistent with most other tasks. I would recommend `#!/usr/bin/env bash`. This applies to other steps as well. I can see the mocks file already has it.

ligangty replied:
Will fix.

ligangty replied:
Fixed in new commit.

mmalina replied:
Added a new comment here: https://github.com/konflux-ci/release-service-catalog/pull/510/files#r1753909010

mmalina commented on tasks/publish-to-mrrc/publish-to-mrrc.yaml:
```diff
@@ -0,0 +1,59 @@
+        mrrc_dir="$(dirname $MRRC_FILE)/mrrc"
+        mkdir -p $mrrc_dir
+        cd $mrrc_dir
+        oras pull $MRRC_ZIP_REGISTRY
```
So this is all happening in the shared workspace, right? I think that might be problematic, because the PVC is only something like 1gb. Normally I would recommend using a temp dir (preferably via `mktemp -d`). So either you can use that and have just one step. Or else you can use an emptyDir workspace in the Pod and then you can access it in both steps. See for example here: https://github.com/konflux-ci/release-service-catalog/blob/7f1b6ba9ee108d1cd9973b1d6ab9eb182fa3aff3/tasks/push-rpm-manifests-to-pyxis/push-rpm-manifests-to-pyxis.yaml#L35

ligangty replied:
Ok. Will look at it.

ligangty replied:
Changed accordingly in new commit.

mmalina commented on tasks/publish-to-mrrc/publish-to-mrrc.yaml:
```diff
@@ -0,0 +1,59 @@
+        cd $mrrc_dir
+        oras pull $MRRC_ZIP_REGISTRY
+    - name: upload-maven-repo
+      image: quay.io/konflux-ci/charon:b99585182f6750b3f6352889fe9c2222b57833ef
```
Now that this image has been made public, I wonder if it would make sense to simply include charon in the utils image, like we include e.g. oras: https://github.com/konflux-ci/release-service-utils/blob/main/Dockerfile#L19

Or maybe that wouldn't be as simple as that as there might be more things and not just one binary?

Just wanted to ask what you think of this. Using the charon image should be fine as well, especially since it's hosted in the quay.io/konflux-ci namespace.

ligangty replied:
The charon tool is developed using python which has lots of dependencies, so it's a little hard to include it in release-service-utils image. This is the Dockerfile for it: https://github.com/Commonjava/charon/blob/konflux/image/Containerfile
I think if we want to include it, we need to add bunch of changes in the release-service-utils. And there is an undetermined requirement which may need some other tools to do "detached signatures"  in next version. I'm not sure any new changes to the image, so maybe we should leave it single now.

johnbieren replied:
I am fine with it not being in our image. However, we don't like to have images that we don't "control" because if someone goes and deletes it, our pipelines break and we are on the hook for it. I am fine with this being merged like this, but can you file a ticket on the RELEASE Jira board for us to copy the image listed here to one of our quay orgs and update this line?

ligangty replied:
Ok, I'll do this. Do we have a template or sample ticket for me to reference?

mmalina replied:
> The charon tool is developed using python which has lots of dependencies

Yeah, then I agree it makes sense to use a separate image.

ligangty replied:
The ticket is created for tracking.

mmalina replied:
Here's the ticket for reference: https://issues.redhat.com/browse/RELEASE-1154

johnbieren commented on tasks/publish-to-mrrc/publish-to-mrrc.yaml:
```diff
@@ -0,0 +1,59 @@
+        mrrc_dir="$(dirname $MRRC_FILE)/mrrc"
+        mkdir -p $mrrc_dir
+        cd $mrrc_dir
+        oras pull $MRRC_ZIP_REGISTRY
```
oras has been giving us lots of issues because it isn't great at figuring out auth. I think you will need to do something like this https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/create-pyxis-image/create-pyxis-image.yaml#L118

ligangty replied:
Ok. Will check.

ligangty replied:
Changed accordingly in new commit.

johnbieren commented on tasks/publish-to-mrrc/README.md:
```diff
@@ -0,0 +1,12 @@
+| Name               | Description                                                                                                                                     | Optional | Default value |
+|--------------------|-------------------------------------------------------------------------------------------------------------------------------------------------|----------|---------------|
+| mrrcParamFilePath  | Path of the mrrc.env file which contains the MRRC parameters as environment viariables                                                          | No       | -             |
+| cosignPubKeyConfig | The configmap which contains the public key to do the cosign signature and attestation check of maven zip file. Uses "consign.pub" as file key. | No       | -             |
```
I know there is a thread in the yaml, but opening one here to remove this line

ligangty replied:
Sure. Will remove.

ligangty replied:
Removed in new commit.

johnbieren reviewed (dismissed):
Marking my approval, but let's give Martin a chance to review again and approve before merging

ligangty commented:
@johnbieren Thanks for reviewing. Let's wait for @mmalina 's double check.

mmalina commented on tasks/publish-to-mrrc/tests/mocks.sh:
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
```
I asked for shebangs to be added in the actual task which you did. But I see you added `#!/usr/bin/env sh` and so it's different from the mock now. They should use the same so that the test is as realistic as possible - otherwise the test will run with `bash` and in production it will run with `sh`.

Now which one to use: We use `bash` in most places because otherwise shellcheck will complain about some things like using `<<<`. But if everything works for you with `sh`, feel free to use that one.

ligangty replied:
Um... I remember I've set to bash but seems there is something wrong with my memory.
Whatever I've changed to bash in new commit.  Sorry for this.

mmalina replied:
ok, no worries, thanks

ligangty commented:
@johnbieren @mmalina Seems all reviews done, can we merge this so I can give other PR based on this one?

johnbieren commented:
Can you rebase? Then we can merge

ligangty commented:
@johnbieren Rebased.

johnbieren reviewed (approved):
/ok-to-test

johnbieren commented:
Can you sign the commit? It is one of our repo requirements https://github.com/konflux-ci/release-service/blob/main/CONTRIBUTING.md#signing-commits

ligangty commented:
> Can you sign the commit? It is one of our repo requirements https://github.com/konflux-ci/release-service/blob/main/CONTRIBUTING.md#signing-commits

Oh, sorry. Added commit signing.
